sniffio/_impl.py

```diff
@@ -1,12 +1,27 @@
 from contextvars import ContextVar
-from typing import Optional
+from typing import Optional, TYPE_CHECKING
 import sys
 
+if TYPE_CHECKING or sys.version_info >= (3, 7):
+    import threading
+else:
+    try:
+        import threading
+    except ImportError:  # pragma: no cover
+        import dummy_threading as threading  # pragma: no cover
```

Comment:
twisted works without the `threading` module available, as long as you don't do anything needed threads:

```python
from twisted.internet import task

import sys


sys.modules["threading"] = None  # you don't need threads anyway good riddance


async def foo(reactor):
    while True:
        print("hello")
        await task.deferLater(reactor, 0.5)


task.react(foo)
```

Reply:
It seems to me that `import dummy_threading as threading` is going to fail every time since I am not aware of anywhere that it would be found.  Also, whatever `threading` ends up being it needs to have a `.local` attribute that can be inherited from for the sake of this present code.  So, where would `dummy_threading` come from?

Reply:
https://docs.python.org/3.6/library/dummy_threading.html

Reply:
We could use:

```python
if TYPE_CHECKING or sys.version_info >= (3, 7):
    from threading import local as _local
else:
    try:
        from threading import local as _local
    except ImportError:  # pragma: no cover
        _local = object
```

instead